architecture-model-generator/modules/architecture-model-generator-core/src/main/java/io/ballerina/projectservice/core/MultiRootMigrationUtil.java

```diff
@@ -0,0 +1,121 @@
+ */
+
+package io.ballerina.projectservice.core;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Utility class for handling multi-root migration results. Provides common functionality for splitting and organizing
+ * migration results across multiple projects.
+ *
+ * @since 1.5.0
+ */
+public class MultiRootMigrationUtil {
+
+    /**
+     * Processes multiRoot migration results by splitting text edits per project.
+     *
+     * @param result The tool execution result containing all text edits
+     * @return A list of ProjectMigrationResult objects, one per project
+     */
+    public static List<ProjectMigrationResult> processMultiRootResults(ToolExecutionResult result) {
+        Map<String, ProjectMigrationResult> projectResults = new TreeMap<>();
+        Map<String, String> allTextEdits = result.textEdits();
+
+        // Split text edits by project
+        for (Map.Entry<String, String> entry : allTextEdits.entrySet()) {
+            String filePath = entry.getKey();
+            String content = entry.getValue();
+
+            // Check if the file belongs to a project (has a project prefix)
+            String projectName = extractProjectName(filePath);
+
+            if (projectName != null) {
+                // File belongs to a project
+                String relativeFilePath = filePath.substring(projectName.length() + 1); // Remove "projectName/"
+                ProjectMigrationResult projectResult =
+                        projectResults.computeIfAbsent(projectName, ProjectMigrationResult::new);
+
+                // Handle migration report files separately
+                if (relativeFilePath.equals("migration_report.html")) {
+                    projectResult.setReport(content);
+                } else {
+                    projectResult.addTextEdit(relativeFilePath, content);
+                }
+            }
+        }
+
+        // Return sorted list of results
+        return new ArrayList<>(projectResults.values());
+    }
+
+    /**
+     * Extracts the project name from a file path in multiRoot format. For example: "project1/main.bal" -> "project1",
+     * "Ballerina.toml" -> null
+     *
+     * @param filePath The file path to parse
+     * @return The project name if the path has a project prefix, null otherwise
+     */
+    public static String extractProjectName(String filePath) {
+        int slashIndex = filePath.indexOf('/');
+        if (slashIndex <= 0) {
+            // No slash or slash at the beginning - this is a root-level file
+            return null;
+        }
+
+        String possibleProjectName = filePath.substring(0, slashIndex);
+        // Exclude common report files that shouldn't be treated as project names
+        if (possibleProjectName.contains(".") || possibleProjectName.contains("_report")) {
+            return null;
+        }
+
+        return possibleProjectName;
+    }
+
```

Comment:
This implementation is fine for now, but we should revisit this later to use the project API. We cannot always assume the project name would be the directory name.